File: cctbx/xray/boost_python/gradient_flags.cpp
#include <boost/python/class.hpp>
#include <cctbx/xray/gradient_flags.h>

namespace cctbx { namespace xray { namespace boost_python {

namespace {

  struct gradient_flags_wrappers
  {
    typedef gradient_flags w_t;

    static void
    wrap()
    {
      using namespace boost::python;
      class_<w_t>("gradient_flags", no_init)
        .def(init<bool, bool, bool, bool, bool, bool>())
        .def_readwrite("site", &w_t::site)
        .def_readwrite("u_iso", &w_t::u_iso)
        .def_readwrite("u_aniso", &w_t::u_aniso)
        .def_readwrite("occupancy", &w_t::occupancy)
        .def_readwrite("fp", &w_t::fp)
        .def_readwrite("fdp", &w_t::fdp)
        .def("all_false", &w_t::all_false)
        .def("adjust", &w_t::adjust)
      ;
    }
  };

} // namespace <anoymous>

  void wrap_gradient_flags()
  {
    gradient_flags_wrappers::wrap();
  }

}}} // namespace cctbx::xray::boost_python

File: cctbx/cctbx/xray/structure_factors/gradient_flags.py
from cctbx.xray import ext
import boost.python
import sys

class gradient_flags(ext.gradient_flags):

  def __init__(self, site=None,
                     u_iso=None,
                     u_aniso=None,
                     occupancy=None,
                     fp=None,
                     fdp=None,
                     u=None,
                     default=False):
    if (u is not None): assert u_iso is None and u_aniso is None
    if (u is None): u = default
    if (site is None): site = default
    if (u_iso is None): u_iso = u
    if (u_aniso is None): u_aniso = u
    if (occupancy is None): occupancy = default
    if (fp is None): fp = default
    if (fdp is None): fdp = default
    ext.gradient_flags.__init__(self, site, u_iso, u_aniso, occupancy, fp, fdp)

class _gradient_flags(boost.python.injector, ext.gradient_flags):

  def copy(self):
    return ext.gradient_flags(self)

  def show_summary(self, f=None):
    if (f is None): f = sys.stdout
    print >> f, "gradient_flags:"
    print >> f, "  site:", self.site
    print >> f, "  u_iso:", self.u_iso
    print >> f, "  u_aniso:", self.u_aniso
    print >> f, "  occupancy:", self.occupancy
    print >> f, "  fp:", self.fp
    print >> f, "  fdp:", self.fdp
